File: fizzbuzz.rb
def fizzbuzz(i)
    return "FizzBuzz!" if i % 15 == 0
    return "Fizz!" if i % 3 == 0
    return "Buzz!" if i % 5 == 0
    i.to_s
end

(1..100).each { |i| puts fizzbuzz(i) }

File: fizzbuzz.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub fizzbuzz {
    my ($i) = @_;
    return "Fizz,Buzz" if $i % 15 == 0;
    return "Fizz" if $i % 3 == 0;
    return "Buzz" if $i % 5 == 0;
    return $i;
}

print fizzbuzz($_) . "\n" for 1..100;
